lib/middleware.js

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright (c) 2020-2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import bedrock from 'bedrock';
+import 'bedrock-express';
+import onHeaders from 'on-headers';
+
+bedrock.events.on('bedrock-express.init', app => {
+  app.use((req, res, next) => {
+    onHeaders(res, function() {
+      /* Note: This fix is for Safari browsers before version 14.1.x. Those
+      browsers did not handle the cookie attribute and value `SameSite=None`.
+      They either treated it as `SameSite=Strict` or broke in other ways. The
+      fix is to not setthat attribute for those browsers. That fix is
+      implemented here. */
+      const setCookie = res.getHeader('set-cookie');
+      if(!setCookie) {
+        return;
+      }
+      const userAgent = req.get('user-agent');
+      if(!(_hasSafari(userAgent) && _isOldSafariVersion(userAgent))) {
+        return;
+      }
+      let hasSameSiteNone = false;
+      if(!Array.isArray(setCookie)) {
+        hasSameSiteNone = _hasSameSiteNone(setCookie);
+        if(hasSameSiteNone) {
+          const newVal = setCookie.replace(' SameSite=None', '');
+          res.setHeader('set-cookie', newVal);
+        } else {
+          res.setHeader('set-cookie', setCookie);
+        }
+      } else {
+        const cookies = [];
+        for(const value of setCookie) {
+          hasSameSiteNone = _hasSameSiteNone(value);
+          if(hasSameSiteNone) {
+            const newVal = value.replace(' SameSite=None', '');
+            cookies.push(newVal);
+          } else {
+            cookies.push(value);
+          }
+        }
+        res.setHeader('set-cookie', cookies);
+      }
```

Comment:
There's no need to set the header again if it doesn't have `SameSite=None`:

```suggestion
      if(!Array.isArray(setCookie)) {
        if(!_hasSameSiteNone(setCookie)) {
          return;
        }
        const newVal = setCookie.replace('SameSite=None', '');
        res.setHeader('set-cookie', newVal);
      } else {
        let updated = false;
        for(const [i, value] of setCookie.entries()) {
          if(!_hasSameSiteNone(value)) {
            continue;
          }
          setCookie[i] = value.replace('SameSite=None', '');
          updated = true;
        }
        if(updated) {
          res.setHeader('set-cookie', setCookie);
        }
      }
```